tingbot/graphics.py

```diff
@@ -217,6 +217,51 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
 
         self.image(text_image, xy=xy, align=align, scale=1)
 
+    def oval(self, xy=None, size=(100,100), color='grey', align='center'):
+        """
+        Draws an oval.
+        
+        Args:
+            xy (tuple): The position (x, y) to draw the oval, as measured from the top-left.
+            size (tuple): The size (width, height) of the oval.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the oval relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if len(size) != 2:
+            raise ValueError('size should be a 2-tuple')
+
+        xy = _topleft_from_aligned_xy(xy, align, size, self.size)
+
+        pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size), 0) 
+
+    def circle(self, xy=None, size=100, color='grey', align='center'):
+        """
+        Draws a circle.
+
+        Args:
+            xy (tuple): The position (x, y) to draw the circle, as measured from the top-left.
+            size (int): The diameter of the circle.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the circle relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if xy is None:
+            raise ValueError('no value given for xy')
+        elif type(xy) is not tuple:
+            raise ValueError('xy should be a 2-tuple')
+        elif len(xy) != 2:
+            raise ValueError('xy should be a 2-tuple')
```

Comment:
Sorry, I wasn't clear - the code I posted above is what I'd consider acceptable parameter checking, since it doesn't use isinstance to check a type hierarchy, it only checks what an object can actually _do_ (this is called duck typing). So I'd like to keep some parameter checking using the code above.